File: test_nyc_integration.py
#!/usr/bin/env python3
"""
Test script for NYC compliance system integration
"""

import asyncio
import json
from complianceNYC import ComprehensivePropertyComplianceSystem

async def test_nyc_compliance():
    """Test the comprehensive NYC compliance system"""
    
    print("🧪 Testing NYC Compliance System Integration")
    print("=" * 60)
    
    # Initialize the comprehensive compliance system
    compliance_system = ComprehensivePropertyComplianceSystem()
    
    # Test addresses
    test_addresses = [
        "140 West 28th Street, New York, NY 10001",
        "666 Broadway, New York, NY 10012",
        "1 Wall Street, New York, NY 10005"
    ]
    
    for address in test_addresses:
        print(f"\n🏢 Testing address: {address}")
        print("-" * 40)
        
        try:
            # Process the property
            record = await compliance_system.process_property(address)
            
            # Display results
            print(f"✅ Address: {record.address}")
            print(f"   BIN: {record.bin}")
            print(f"   BBL: {record.bbl}")
            print(f"   Borough: {record.borough}")
            print(f"   Overall Score: {record.overall_compliance_score:.1f}/100")
            print(f"   HPD Violations: {record.hpd_violations_total} total, {record.hpd_violations_active} active")
            print(f"   DOB Violations: {record.dob_violations_total} total, {record.dob_violations_active} active")
            print(f"   Elevator Devices: {record.elevator_devices_total} total, {record.elevator_devices_active} active")
            print(f"   Boiler Devices: {record.boiler_devices_total} total")
            print(f"   Electrical Permits: {record.electrical_permits_total} total, {record.electrical_permits_active} active")
            
            # Show sample violations if available
            hpd_violations = json.loads(record.hpd_violations_data)
            if hpd_violations:
                print(f"   📋 Sample HPD Violation: {hpd_violations[0].get('novdescription', 'N/A')[:50]}...")
            
            dob_violations = json.loads(record.dob_violations_data)
            if dob_violations:
                print(f"   📋 Sample DOB Violation: {dob_violations[0].get('violation_type', 'N/A')}")
            
            print(f"   📊 Data Sources: {record.data_sources}")
            print(f"   ⏰ Processed: {record.processed_at}")
            
        except Exception as e:
            print(f"❌ Error processing {address}: {e}")
    
    print(f"\n✅ NYC Compliance System Integration Test Complete")
    print("=" * 60)

if __name__ == "__main__":
    asyncio.run(test_nyc_compliance())

File: test_nyc_integration.js
#!/usr/bin/env node
/**
 * Test script to verify NYC data integration
 * This script tests the Python compliance system integration
 */

const { spawn } = require('child_process');
const path = require('path');

async function testNYCIntegration() {
  console.log('🧪 Testing NYC Data Integration');
  console.log('=' * 50);
  
  const testAddress = '140 West 28th Street, New York, NY 10001';
  
  try {
    console.log(`📍 Testing address: ${testAddress}`);
    console.log('🐍 Running Python compliance script...');
    
    const pythonProcess = spawn('python3', [
      './complianceNYC.py',
      testAddress,
      'Manhattan'
    ]);

    let output = '';
    let errorOutput = '';

    pythonProcess.stdout.on('data', (data) => {
      const text = data.toString();
      output += text;
      console.log('Python output:', text);
    });

    pythonProcess.stderr.on('data', (data) => {
      const text = data.toString();
      errorOutput += text;
      console.error('Python error:', text);
    });

    pythonProcess.on('close', (code) => {
      console.log(`\n🐍 Python process exited with code: ${code}`);
      
      if (code === 0) {
        console.log('✅ Python script completed successfully');
        
        // Try to parse JSON output
        try {
          const lines = output.split('\n');
          let jsonOutput = '';
          let inJsonSection = false;
          
          for (const line of lines) {
            if (line.includes('{') && line.includes('"address"')) {
              inJsonSection = true;
            }
            if (inJsonSection) {
              jsonOutput += line + '\n';
              if (line.includes('}') && line.includes('"data_sources"')) {
                break;
              }
            }
          }
          
          if (jsonOutput.trim()) {
            const complianceData = JSON.parse(jsonOutput.trim());
            console.log('📊 Compliance Data Summary:');
            console.log(`   Address: ${complianceData.address}`);
            console.log(`   BIN: ${complianceData.bin}`);
            console.log(`   BBL: ${complianceData.bbl}`);
            console.log(`   Borough: ${complianceData.borough}`);
            console.log(`   Overall Score: ${complianceData.overall_compliance_score}/100`);
            console.log(`   HPD Violations: ${complianceData.hpd_violations_total} total, ${complianceData.hpd_violations_active} active`);
            console.log(`   DOB Violations: ${complianceData.dob_violations_total} total, ${complianceData.dob_violations_active} active`);
            console.log(`   Elevator Devices: ${complianceData.elevator_devices_total} total, ${complianceData.elevator_devices_active} active`);
            console.log(`   Boiler Devices: ${complianceData.boiler_devices_total} total`);
            console.log(`   Electrical Permits: ${complianceData.electrical_permits_total} total, ${complianceData.electrical_permits_active} active`);
            
            console.log('\n✅ NYC Data Integration Test PASSED');
            console.log('🎯 Real NYC data is being fetched successfully!');
          } else {
            console.log('⚠️ No JSON output found in Python script');
            console.log('📝 Full output:', output);
          }
        } catch (parseError) {
          console.error('❌ Error parsing Python output:', parseError);
          console.log('📝 Raw output:', output);
        }
      } else {
        console.error(`❌ Python script failed with code ${code}`);
        console.error('Error output:', errorOutput);
        console.log('📝 Full output:', output);
      }
    });

    pythonProcess.on('error', (error) => {
      console.error('❌ Failed to start Python process:', error);
    });

  } catch (error) {
    console.error('❌ Test failed:', error);
  }
}

// Run the test
testNYCIntegration().catch(console.error);
